python_modules/dagster/dagster_tests/definitions_tests/declarative_automation_tests/daemon_tests/definitions/backfill_with_runs_and_checks.py

```diff
@@ -0,0 +1,76 @@
+import dagster as dg
+
+static_partitions1 = dg.StaticPartitionsDefinition(["x", "y", "z"])
+static_partitions2 = dg.StaticPartitionsDefinition(["t", "u", "v"])
+
+condition = dg.AutomationCondition.missing() & ~dg.AutomationCondition.in_progress()
+
+
+@dg.asset(
+    automation_condition=condition,
+    check_specs=[dg.AssetCheckSpec(name="inside", asset="backfillA")],
+)
+def backfillA() -> dg.MaterializeResult:
+    return dg.MaterializeResult(check_results=[dg.AssetCheckResult(passed=True)])
+
+
+@dg.asset(
+    deps=[backfillA],
+    automation_condition=condition,
+    partitions_def=static_partitions2,
+    check_specs=[dg.AssetCheckSpec(name="inside", asset="backfillB")],
+)
+def backfillB() -> dg.MaterializeResult:
+    return dg.MaterializeResult(check_results=[dg.AssetCheckResult(passed=True)])
+
+
+@dg.asset(
+    deps=[backfillA],
+    automation_condition=condition,
+    partitions_def=static_partitions1,
+)
+def backfillC() -> None: ...
+
+
+@dg.asset(
+    automation_condition=condition,
+    check_specs=[dg.AssetCheckSpec(name="inside", asset="run1")],
+)
+def run1() -> dg.MaterializeResult:
+    return dg.MaterializeResult(check_results=[dg.AssetCheckResult(passed=True)])
+
+
+@dg.asset(
+    automation_condition=condition,
+    partitions_def=static_partitions1,
+    check_specs=[dg.AssetCheckSpec(name="inside", asset="run2")],
+)
+def run2() -> dg.MaterializeResult:
+    return dg.MaterializeResult(check_results=[dg.AssetCheckResult(passed=True)])
+
+
+@dg.asset_check(name="outside", asset="backfillA")
+def outsideA() -> dg.AssetCheckResult: ...
```

Comment:
[1] I dont see these `outside` asset checks used in any of the tests

Reply:
see below comment, basic idea is just that we're making sure that these don't tag along for the ride or otherwise cause issues in run construction. I'll add a comment to that effect.